raimitigations/automitigator/evaluator.py

```diff
@@ -0,0 +1,208 @@
+import math
+import traceback
+import raimitigations.dataprocessing as dp
+from flaml import AutoML
+from imblearn.pipeline import Pipeline
+
+from .mitigation_actions import MitigationActions
+
+
+class Evaluator:
+    """
+    Evaluates a given set of mitigations on a given dataset.
+    """
+
+    def __init__(self, automl_args=None) -> None:
+        self.automl_args = automl_args
+        self.pipeline = None
+
+    def _pipeline_append(self, step):
+        """
+        Append a step to the pipeline.
+        Note: This is done to support older versions of sklearn < 1.1 (required to support python 3.7)
+        which doesn't support an empty pipeline to be initialized.
+
+        :param sklearn.pipeline.Pipeline pipeline: The pipeline to add the step to
+        :param step: The step to add
+        """
+        if self.pipeline is None:
+            self.pipeline = Pipeline([step])
+        else:
+            self.pipeline.steps.append(step)
+
+    def evaluate(self, train_x, train_y, search_config):
+        """
+        Evaluates a given set of mitigations on a given dataset.
+
+        :param train_x: The training data
+        :param train_y: The training labels
+        :param dict search_config: The search configuration
+
+        :return: The results of the evaluation
+        :rtype: dict
+        """
+        # Sample config
+        # config: {'search_space':
+        #           {'cohort': 'all',
+        #            'mitigations':
+        #               {'action0': {'type': 0, 'strategy': 0, 'name': 'rebalancer'}}}}
+
+        self.pipeline = None
+        search_space = search_config["search_space"]
+        cohort = search_space["cohort"]
+        if cohort == "all":
+            return self.mitigate_full_dataset(train_x, train_y, search_space)
+        else:
+            raise ValueError(f"Unknown cohort type {cohort}")
+
+    def _process_feature_selector(self, selector_type):
+        """
+        Process the feature selector
+
+        :param dict selector_type: The feature selector configuration
+
+        :raises ValueError: If the feature selector is unknown
+        """
+        selector_name = selector_type["selector_name"]
```

Comment:
Usually, I like to define these strings in some sort of class with all relevant constants. This way, if we ever need to change one of these values, we can do that easily by changing only one value. For example, if you decide to change from '``selector_name``' to '``name_selector``', you'll need to do a search and replace for all '``selector_name``' occurrences. But if you have all constants in a single place, you only need to change it there.

Reply:
Fixed.